source/common/access_log/utility.cc

```diff
@@ -0,0 +1,64 @@
+#include "source/common/access_log/utility.h"
+
+#include <string>
+
+#include "envoy/access_log/access_log.h"
+
+#include "source/common/common/empty_string.h"
+
+namespace Envoy {
+namespace AccessLog {
+
+const std::string& Utility::getAccessLogTypeString(const AccessLogType access_log_type) {
+  switch (access_log_type) {
+  case AccessLogType::TcpUpstreamConnected:
+    return AccessLogTypeStrings::get().TcpUpstreamConnected;
+  case AccessLogType::TcpPeriodic:
+    return AccessLogTypeStrings::get().TcpPeriodic;
+  case AccessLogType::TcpEnd:
+    return AccessLogTypeStrings::get().TcpEnd;
+  case AccessLogType::DownstreamStart:
+    return AccessLogTypeStrings::get().DownstreamStart;
+  case AccessLogType::DownstreamPeriodic:
+    return AccessLogTypeStrings::get().DownstreamPeriodic;
+  case AccessLogType::DownstreamEnd:
+    return AccessLogTypeStrings::get().DownstreamEnd;
+  case AccessLogType::UpstreamStart:
+    return AccessLogTypeStrings::get().UpstreamStart;
+  case AccessLogType::UpstreamPeriodic:
+    return AccessLogTypeStrings::get().UpstreamPeriodic;
+  case AccessLogType::UpstreamEnd:
+    return AccessLogTypeStrings::get().UpstreamEnd;
+  default:
```

Comment:
case `AccessLogType::NotSet:` would be better than `default`.

Reply:
There's a small problem doing so because the way protoc compiles the .proto file causes all the enums in the package to be under the package namespace, that causes a compile time error if there are unhandled enum cases here, even if they are not related to the ``AccessLogType`` enum. So for compilation I have to use the ``default`` keyword